src/components/Buttons/RadioCheckbox.js

```diff
@@ -0,0 +1,147 @@
+import React, { Component, PropTypes }        from 'react'
+import Radium                                 from 'radium'
+import svgRadioBtnBase                        from '../../assets/radioBtnBase.svg'
+import svgRadioBtnChecked                     from '../../assets/radioBtnChecked.svg'
+import svgCheckboxBase                        from '../../assets/checkboxBase.svg'
+import svgCheckboxChecked                     from '../../assets/checkboxChecked.svg'
+
+const BASE_STYLES = {
+  input: {
+    margin: '0 10px -7px 0',
+    backgroundSize: 'contain',
+    backgroundRepeat: 'no-repeat',
+    backgroundPosition: 'top left',
+    backgroundColor: 'transparent',
+    appearance: 'none',
+    '-moz-appearance': 'none',
+    '-webkit-appearance': 'none',
+    width: 22,
+    height: 22,
+  },
+}
+
+const STYLES = {
+  radioBase: {
+    ...BASE_STYLES.input,
+    backgroundImage: `url(${svgRadioBtnBase.url})`,
+    ':active': {
+      backgroundImage: `url(${svgRadioBtnBase.url})`,
+      border: '0 solid transparent'
+    }
+  },
+  radioChecked: {
+    backgroundImage: `url(${svgRadioBtnChecked.url})`,
+    ':active': {
+      backgroundImage: `url(${svgRadioBtnChecked.url})`,
+      border: '0 solid transparent'
+    }
+  },
+  checkboxBase: {
+    ...BASE_STYLES.input,
+    backgroundImage: `url(${svgCheckboxBase.url})`,
+    ':active': {
+      backgroundImage: `url(${svgCheckboxBase.url})`,
+      border: '0 solid transparent'
+    }
+  },
+  checkboxChecked: {
+    backgroundImage: `url(${svgCheckboxChecked.url})`,
+    ':active': {
+      backgroundImage: `url(${svgCheckboxChecked.url})`,
+      border: '0 solid transparent'
+    }
+  }
+}
+
+class RadioCheckbox extends Component {
+  static propTypes = {
+    aria          : PropTypes.shape({
+      label         :PropTypes.string,
+    }),
+    btnType       : PropTypes.oneOf(['radio', 'checkbox']),
+    id            : PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+    isChecked     : PropTypes.bool,
+    onClick       : PropTypes.func,
+    styles        : PropTypes.shape({
+      button           : PropTypes.object,
+    }),
+    value         : PropTypes.string,
+  }
+
+  static defaultProps = {
+    isChecked   : false,
+    btnType     : 'radio',
+    styles      : {},
+    aria        : {},
+  }
+
+  state = {
+    isChecked   : this.props.isChecked,
+    styles      : {}
+  }
+
+  componentDidMount() {
+    this.onMount()
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.isChecked !== prevState.isChecked) {
+      this.setState({styles: this.getStyles()})
+    }
+
+    if (this.props.isChecked !== prevProps.isChecked) {
+      this.setState({isChecked: this.props.isChecked})
+    }
+  }
+
+  onMount() {
+    this.setState({styles: this.getStyles()})
```

Comment:
this method doesn't seem complex enough to be abstracted out from `componentDidMount`.

Reply:
This reason for this comes from the linter in the `carrot` repo:

![screen shot 2017-10-25 at 5 01 33 pm](https://user-images.githubusercontent.com/9634729/32029037-b06a97b8-b9a6-11e7-804c-621d5f4714ea.png)

The way I interpret this rule is to avoid firing various `setState` calls in one method. I got used to it and I always now take this approach. However, now that you bring it up I noticed that this rule is not turned on in Snacks.

What do you think, should I not follow that rule here?

This comment also addresses https://github.com/instacart/Snacks/pull/70/files#r147008676 @wfro

Reply:
Ah, right. This is to avoid setting state immediately on mount, which causes an immediate re-render. This should instead go in componentWillMount. This way only one render is occurs on mount.

Reply:
Any thoughts from https://github.com/instacart/Snacks/pull/70/files#r147008560? Unless I'm missing something (very possible :))I think we can drop calls to `this.setState({ styles: this.getStyles() })` entirely

Reply:
Yeah, I don't think there is any reason to store styles in state